examples/server/grammar.py

```diff
@@ -0,0 +1,99 @@
+from openai import OpenAI
+
+client = OpenAI(api_key="foobar", base_url="http://localhost:1234/v1/")
+
+JSON_KBNF = '''
+(* JSON Grammar *)
+
+(* JSON text must contain a single JSON value *)
+start       = value ;
+
+(* A JSON value can be an object, array, string, number, true, false, or null *)
+value      ::= object 
+           | array 
+           | string 
+           | number 
+           | "true" 
+           | "false" 
+           | "null" ;
+
+(* A JSON object is a collection of key/value pairs enclosed in curly braces *)
+object     ::= "{" [ members ] "}" ;
+members    ::= pair { "," pair } ;
+pair       ::= string ":" value ;
+
+(* A JSON array is an ordered list of values enclosed in square brackets *)
+array      ::= "[" [ elements ] "]" ;
+elements   ::= value { "," value } ;
+
+(* A JSON string is a sequence of Unicode characters enclosed in double quotes *)
+string     ::= "\"" { character } "\"" ;
```

Comment:
You can use regex in kbnf, see https://docs.rs/kbnf/latest/kbnf/#regular-expression.